src/MoBi.Presentation/Presenter/ChartPresenter.cs

```diff
@@ -274,14 +313,17 @@ public virtual void UpdateTemplatesBasedOn(IWithChartTemplates withChartTemplate
          ResetMenus();
       }
 
-      private void addDataRepositorySimulation(DataRepository dataRepository)
+      private void addDataRepositoriesToSimulation(IReadOnlyCollection<DataRepository> dataRepositories)
       {
-         if (_simulations.Contains(dataRepository))
-            return;
+         dataRepositories.Each(dataRepository =>
+         {
+            if (_simulations.Contains(dataRepository))
```

Comment:
use a where condition before the Each. `dataRepositories.Where(x=>!_simulation.Contains(x)).Each`